lib/models.go

```diff
@@ -207,7 +207,7 @@ type Check struct {
 	Fails  int64 `json:"fails"`
 }
 
-// NewCheck creates a new check with the given name and parent group. The group may not be nil.
+// NewCheck creates a new check with the given name and parent group. The group must not be nil.
```

Comment:
Good catch! Keeping in mind comment, maybe add a validation to constructor with explicit error? :thinking:

Reply:
Done @ https://github.com/grafana/k6/pull/4089/commits/58e3d021ea7509bfa7aba35b0babefb7c54e04cc